crates/net/network/src/network.rs

```diff
@@ -435,6 +444,8 @@ struct NetworkInner<N: NetworkPrimitives = EthNetworkPrimitives> {
     discv4: Option<Discv4>,
     /// The instance of the discv5 service
     discv5: Option<Discv5>,
+    /// Sender for basic peer lifecycle events.
+    peer_events_sender: EventSender<PeerEvent>,
```

Comment:
oh I think this is likely not necessary because peerEvent is a subset of of NetworkEvent now, so we can use the same event channel if we change the stream type